aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -112,12 +112,65 @@ static void cumprod_cpu_kernel(const Tensor& result, const Tensor& self, int64_t
     );
   });
 }
+// custom min and max to be used in logcumsumexp for complex arguments
+template <typename scalar_t>
+c10::complex<scalar_t> _logcumsumexp_minmax(c10::complex<scalar_t> x, c10::complex<scalar_t> y, bool min) {
+  scalar_t xr = std::real(x);
+  scalar_t yr = std::real(y);
```

Comment:
Why not use `auto` and `real`/`image` methods?
```suggestion
  auto xr = x.real();
  auto yr = y.imag();
```

Reply:
Done in the next commit